### PR TITLE
[plugin.video.mediathek] 0.7.5

### PR DESCRIPTION
**Pull request by raptor2101**
### Description
ZDF Completly changes its WebInterface so the plugin is rewriten to adopt that (now using JSON API, hopefully more stable)

### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### Additional information :
I'am aware that the code provided by this PullRequest pusches 129 CodeIssues (regarding to codacy). I  was unaware of that until now and will solve these issues with the next PR.

### CODE REVIEW

**Comment by raptor2101** [comment]
Are there any Problems with this PR?

**Comment by Razzeee** [comment]
Initial submissions might take a while, as these tend to be more to check.

**Comment by raptor2101** [comment]
Can i support the checks in any way?

File: plugin.video.mediathek/html.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+
+def transformHtmlCodes(string):
+  replacements = (
+    (u'Ä', u'&Auml;'),
```
Comment (phil65) [inline plugin.video.mediathek/html.py]: did you try xml.sax.saxutils.escape()/unescape()?

File: plugin.video.mediathek/mediathek/__init__.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          return doc.encode('utf-8');
+        except:
+          return doc;
+      else:
+        return ''
+    except:
+      return ''
+      
+  def buildMenu(self, path, treeNode = None):
+    if(type(path) in (str,unicode)):
```
Comment (phil65) [inline plugin.video.mediathek/mediathek/__init__.py]: "if isinstance(path, basestring)"

File: plugin.video.mediathek/resources/language/English/strings.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (phil65) [inline plugin.video.mediathek/resources/language/English/strings.xml]: please convert to .po files.

File: plugin.video.mediathek/mediathek/__init__.py (hunk 2 above)
Comment (raptor2101) [inline plugin.video.mediathek/mediathek/__init__.py]: Known, i'am currently merging the hints from codacy into the code ;)

File: plugin.video.mediathek/html.py (hunk 1 above)
Comment (raptor2101) [inline plugin.video.mediathek/html.py]: Known, this lines comes from times of the plugin where this wasn't available or known to me ;)

File: plugin.video.mediathek/mediathek/__init__.py (hunk 2 above)
Comment (raptor2101) [inline plugin.video.mediathek/mediathek/__init__.py]: see here: https://www.codacy.com/app/raptor2101/Mediathek/dashboard

File: plugin.video.mediathek/simplexbmc.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    hashGenerator.update(sys.argv[2]);
+    callhash = hashGenerator.hexdigest();
+    storedJsonFile = os.path.join(self.plugin_profile_dir,"%s.json"%callhash);
+    output = open(storedJsonFile, 'wb');
+    json.dump(jsonObject,output);
+    return callhash;
+  
+  def loadJsonFile(self,callhash):
+    storedJsonFile = os.path.join(self.plugin_profile_dir,"%s.json"%callhash);
+    input = open(storedJsonFile,"rb");
```
Comment (phil65) [inline plugin.video.mediathek/simplexbmc.py]: same here

File: plugin.video.mediathek/simplexbmc.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    return json.load(input);
+
+  def buildJsonLink(self,mediathek,title,jsonPath,callhash,objectCount):
+    listItem=xbmcgui.ListItem(title, iconImage="DefaultFolder.png")
+    
+    url = "%s?type=%s&action=openJsonPath&path=%s&callhash=%s" % (sys.argv[0],mediathek.name(), jsonPath,callhash)
+    xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=url,listitem=listItem,isFolder=True,totalItems = objectCount)
+
+  def listAvaibleMediathekes(self, mediathekNames):
```
Comment (phil65) [inline plugin.video.mediathek/simplexbmc.py]: Avaible -> Available?

**Comment by raptor2101** [comment]
I commit some changes to follow your advises. Do you close these PR or should i do a second commit on try on this PR...

**Comment by phil65** [comment]
just add a second commit here.

**Comment by raptor2101** [comment]
fix pushed

**Comment by phil65** [comment]
Thx! Looks good now.

**Comment by raptor2101** [comment]
Can you look at the parallel PR (https://github.com/xbmc/repo-plugins/pull/757) it's literal the same code only marked as "deprecated" for all gotham based version...

**Pull request merged**